BankSDK/GiniBankSDKExample/GiniBankSDKShareExtension/ShareViewController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ShareViewController.swift
+//  GiniBankSDKShareExtension
+//
+//  Created by Nadya Karaban on 13.10.22.
+//
+
+import UIKit
+import Social
+import CoreServices
+
+class ShareViewController: UIViewController {
+    private let typeURL = String(kUTTypeImage)
+    private let appURL = "BankSDKExtension://"
+    private let groupName = "group.bank.extension.test"
+    private let urlDefaultName = "incomingURL"
+    private let imageKey = "imageData"
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        guard let extensionItem = extensionContext?.inputItems.first as? NSExtensionItem,
+            let itemProvider = extensionItem.attachments?.first else {
+                self.extensionContext?.completeRequest(returningItems: nil, completionHandler: nil)
+                return
+        }
+        if itemProvider.hasItemConformingToTypeIdentifier(typeURL) {
+            handleIncomingURL(itemProvider: itemProvider)
+        } else {
+            self.extensionContext?.completeRequest(returningItems: nil, completionHandler: nil)
+        }
+    }
+      
+    private func save(_ data: Data, key: String, value: Any) {
+      // You must use the userdefaults of an app group, otherwise the main app don't have access to it.
+        if let userDefaults = UserDefaults(suiteName: groupName){
+            userDefaults.set(data, forKey: imageKey)
+        }
+    }
+        
+
+    private func handleIncomingURL(itemProvider: NSItemProvider) {
+        itemProvider.loadItem(forTypeIdentifier: typeURL, options: nil) { (item, error) in
+            if let error = error { print("URL-Error: \(error.localizedDescription)") }
+
+            if let url = item as? NSURL, let urlString = url.absoluteString {
+                let imageData = try? Data(contentsOf: url as URL)
+                self.save(imageData!, key: self.imageKey, value: imageData!)
```

Comment:
[Clarification]`imageData` may have a nil value at same point and this can break the app if we are force unwrapping it. I suggest to verify if it has any value before using it.